src/cast_core/src/RunCommand.cpp

```diff
@@ -208,13 +208,110 @@ class DatabaseOperation {
     bool _awaitStepdown;
 };
 
+// Class to parse and hold the OnlyRunInInstance configuration. This option may be used to only
+// run under certain configurations Available configurations are specified in Options.
+class InstanceTypeFilter {
+public:
+    struct Options {
+        static constexpr std::string_view kStandalone = "standalone";
+        static constexpr std::string_view kSharded = "sharded";
+        static constexpr std::string_view kReplicaSet = "replica_set";
+
+        static std::string getValidOptionsListAsString() {
+            std::stringstream listString;
+            listString << kStandalone << ", " << kSharded << ", " << kReplicaSet;
+            return listString.str();
+        }
+
+        static bool isValid(std::string_view instanceType) {
+            return instanceType == kStandalone || instanceType == kSharded
+                    || instanceType == kReplicaSet;
+        }
+
+        static bool isValidList(const std::vector<std::string>& options) {
+            for(auto &option : options) {
+                if(!isValid(option)) {
+                    return false;
+                }
+            }
+            return true;
+        }
+    };
+
+    InstanceTypeFilter(PhaseContext &context, mongocxx::pool::entry& client) {
+        try {
+            _onlyRunInInstancesContext = context.getPlural<std::string>(
+                "OnlyRunInInstance", "OnlyRunInInstances", [&](const Node& node) {
```

Comment:
So there's a singular and plural version? I'd say we just keep the plural version; people can make it a one-item list with `key: [value]` and it both makes this implementation easier and makes the overall syntax simpler.

Reply:
I have seen this singular/plural scheme used in the existing actors, which is why I decided to follow this pattern. Is this no longer the recommended way? I just want to be consistent with the rest of actors.

Reply:
Ah, fair in that case! I don't like it personally but I can deal (and won't block on it!)